Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -86,16 +83,24 @@ public final class RemoteMessagingStore: RemoteMessagingStoring {
         os_log("Remote messaging config - save processed version: %d", log: log, type: .debug, processorResult.version)
 
         let context = database.makeContext(concurrencyType: .privateQueueConcurrencyType, name: Constants.privateContextName)
-        saveRemoteMessagingConfig(withVersion: processorResult.version, in: context)
+        context.performAndWait {
+            storeRemoteMessagingConfig(with: processorResult.version, in: context)
 
-        if let remoteMessage = processorResult.message {
-            deleteScheduledRemoteMessages(in: context)
-            save(remoteMessage: remoteMessage, in: context)
+            if let remoteMessage = processorResult.message {
+                addOrUpdate(remoteMessage: remoteMessage, in: context)
+            } else {
+                markScheduledMessagesAsDone(in: context)
+            }
+            deleteNotShownDoneMessages(in: context)
```

Comment:
If there's a new message to be saved, add or update it. Update, in case that ID has already been shown to the user, in which case we're reusing the database object.

If there's no message in the config, we mark scheduled messages as done.

After that, delete all messages that were previously scheduled and never shown (that includes messages just marked as done in `markScheduledMessagesAsDone`).